src/EVM/Expr.hs

```diff
@@ -954,6 +954,9 @@ simplify e = if (mapExpr go e == e)
 
     -- redundant CopySlice
     go (CopySlice (Lit 0x0) (Lit 0x0) (Lit 0x0) _ dst) = dst
+    go (CopySlice (Lit 0) (Lit 0) (BufLength (AbstractBuf k1))
+      (CopySlice (Lit 0) (Lit 0) (BufLength (AbstractBuf k2)) (AbstractBuf k3) _)
+      (ConcreteBuf "")) | k1 == k2 && k2 == k3 = (AbstractBuf k1)
```

Comment:
```suggestion
    go (CopySlice (Lit 0) (Lit 0) (BufLength (AbstractBuf k))
      (CopySlice (Lit 0) (Lit 0) (BufLength (AbstractBuf k)) (AbstractBuf k) _)
      (ConcreteBuf ""))  = (AbstractBuf k)
```

Also, it feels like this is just a consequence of more general simplification rules.
I will try to write them down here.

Reply:
This... actually doesn't work:

![image](https://github.com/user-attachments/assets/4b3e75e8-4359-45a4-b1fc-826b5f2c9447)

Reply:
It's... a Haskell thing. Don't ask me why :D So the k1..k3 and making them all equal is the only way this can be made to work, AFAIK.

Reply:
However... this inspired me to make it more generic!